### DISCUSSION TITLE
Sveltekit v2 support

### DISCUSSION THREAD

**Discussion by zed-wong** [vercel/community#5208] [Help] [answered]
### Summary

Just upgraded Sveltekit to v2, and the deployment stopped working. 

In `https://vercel.com/{org}/{repo}/settings` -> 'General' -> 'Build & Development Settings' -> 'Framework Preset' there's not an option for sveltekit v2. Hope vercel support it soon.

### Steps to Reproduce

Upgrade sveltekit version to v2, then the build doesn't work anymore.

**Comment by mcmxcdev** [comment]
Hey @zed-wong, did you find a workaround for this?

I am encountering `Error: No Output Directory named "public" found after the Build completed. You can configure the Output Directory in your Project Settings.` with a default setup. Is this the same error you are experiencing?

I was naive to expect that Sveltekit v2 would be supported already since Rich Harris works for Vercel...

**Reply by pboling** [reply]
I was expecting to have the best experience on Vercel, but found this issue, and ended up not even trying Vercel as a result. My Svelte v5.rc + SvelteKit v2.5 app is running great on ~~Heroku~~ (moving to Fly.io to use Electric-SQL)!

It doesn't make any sense.

**Reply by usagizmo** [reply]
I think this error might be caused by turbo v2.
In fact, after reverting turbo from 2.0.1 to 1.13.4, I was able to deploy properly.

**Reply by amyegan** [reply]
Although it wasn't the source of the original issue here, there are some [breaking changes](https://turbo.build/blog/turbo-2-0#breaking-changes) between Turborepo v1 and v2. You can find upgrade information at https://turbo.build/repo/docs/crafting-your-repository/upgrading#update-turbojson

**Comment by amyegan** [comment, marked as answer]
I just created a new SvelteKit v2 project using `npm create svelte@latest` and successfully deployed it with the `SvelteKit (v1)` preset. Could you give that a try and let me know how it goes for your project?

**Reply by mcmxcdev** [reply]
Thanks for looking into this! In my case, it's a Sveltekit v2 project using turborepo.

I had no luck with the default `SvelteKit (v1)` preset (No Output Directory named "build" found after the Build completed error), now trying to resolve this by using a custom `vercel.json` with:

```
{
  "buildCommand": "pnpm turbo build --filter=@org/repo",
  "outputDirectory": "apps/repo/.svelte-kit",
  "installCommand": "pnpm install",
  "devCommand": "pnpm turbo dev --filter=@org/repo",
  "rewrites": [{ "source": "/(.*)", "destination": "/" }]
}
```

Build succeeds, but then I encounter 404 page not found when opening the deployment.

**Reply by mcmxcdev** [reply]
@amyegan any idea what I am doing wrong? Or am I missing some open issue related to integrating Sveltekit with Turborepo?

**Reply by amyegan** [reply]
Our documentation on using monorepos should help you get it set up – https://vercel.com/docs/monorepos

**Comment by pboling** [comment]
Is a default SvelteKit v2 preset planned?
https://svelte.dev/blog/sveltekit-2

**Reply by amyegan** [reply]
Hey @pboling. The `vite` commands stay the same between v1 and v2, so the `SvelteKit (v1)` preset works for both. If you need to customize, you can always override any of the default commands in [Build and Development Settings](https://vercel.com/docs/deployments/configure-a-build#build-and-development-settings)

**Reply by mcmxcdev** [reply]
Then please remove the "v1" indicator from that preset. Developers are used to that indicating that e.g. v1 is not compatible with v2 usages due to possible breaking changes.

**Reply by amyegan** [reply]
Already have a new ticket open for the change 😄
